### ISSUE TITLE
[Bug]: Unable to import Soild's `html` function in Solid Start app

### ISSUE THREAD

**Issue by trusktr** [solidjs/solid-start#1618]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

More details here:

https://github.com/solidjs/solid/issues/1984

### Expected behavior 🤔

All of Solid's packages, including `solid-js` should be importable in Solid Start.

### Steps to reproduce 🕹

Steps:

1. `npm create solid` and follow steps to create a Solid app
2. Add `import html from 'solid-js/html'` to any JS or TS file.
3. Try to run the app and see the error

### Context 🔦

Some components, for example 3rd party components, might be importing Solid's `html`. It would be great for a Solid app with SSR enabled to not have an issue importing such code (there will be more problems to wrangle after importing such as running certain components only on the client, but that's a next step after first getting imports working).

Fixed by 

- https://github.com/ryansolid/dom-expressions/pull/345

**Comment by trusktr** [comment]
Working towards fixing this here:

- https://github.com/ryansolid/dom-expressions/pull/345

**Comment by trusktr** [comment]
I've confirmed after linking modified `solid-js` including this change into a Solid Start app that I'm able to progress towards normal `import` statements.

Even if I'm not calling `html` on the server side, the app is a lot cleaner being able to hoist `import` statements out of components or conditional top-level awaits.

**Comment by trusktr** [comment]
Having normal `import` statements makes things really easyin VS Code too, for example VS Code auto-complete will automatically add `import` statements for anything that you need to import, which is super convenient. VS Code will not automatically add `await import()` anywhere, making it more of a manual process.

**Comment by trusktr** [comment]
Oh my goodness.... YES!

https://github.com/ryansolid/dom-expressions/pull/345 cleans up my Solid Start modules that import `lume`. 

For example I had this ugly code,

<details><summary>ugly code</summary>
<p>

```ts
import {batch, createSignal, onCleanup, onMount} from 'solid-js'
import {element, type ElementAttributes} from '@lume/element'
import {type Element3DAttributes, Element3D, type TextureProjector} from 'lume'
import {createMutable} from 'solid-js/store'
import {createArrayMemo} from '../utils/createArrayMemo.js'

export type TiltCardAttributes = Element3DAttributes | 'image'

// Solid Start import issue: https://github.com/solidjs/solid-start/issues/1614
if (globalThis.window?.document) await defineElement()

async function defineElement() {
	const {default: html} = await import('solid-js/html')
	const {Element3D, Motor, stringAttribute, numberAttribute, booleanAttribute} = await import('lume')
	const {autoDefineElements} = await import('lume/dist/LumeConfig.js')

	return @element('lume-tilt-card', autoDefineElements)
	class TiltCard extends Element3D {
		@numberAttribute rotationAmount = 12
		..etc..

		override template = () => {
			let plane: Element3D

			const state = createMutable({...})

			onMount(() => {
				...
				const task = Motor.addRenderTask(time => {...})
				onCleanup(() => Motor.removeRenderTask(task))
			})

			const [projector, setProjector] = createSignal<TextureProjector>()
			const projectors = createArrayMemo(() => (projector() ? [projector()] : []))

			return html`
				<lume-rounded-rectangle
					id=${() => this.image.split('/').pop()}
					part="root"
					ref=${(e: Element3D) => (plane = e)}
					...
				>
				...
				</lume-rounded-rectangle>

			`
		}

	}
}

// Temporary hack type to get around the async defineElement() workaround
type TiltCardCtor = Awaited<ReturnType<typeof defineElement>>
export type TiltCard = InstanceType<TiltCardCtor>

declare module 'solid-js' {
	namespace JSX {
		interface IntrinsicElements {
			'lume-tilt-card': ElementAttributes<TiltCard, TiltCardAttributes>
		}
	}
}

declare global {
	interface HTMLElementTagNameMap {
		'lume-tilt-card': TiltCard
	}
}
```


</p>
</details> 


Which as you can imagine, gets worse when trying to define more than one custom element in a single file (with JSX types for type checking in Solid components), for example:

<details><summary>even worse code</summary>
<p>

```ts
// ...normal imports...

if (globalThis.window?.document) await defineElement()

async function defineElement() {
	const {default: html} = await import('solid-js/html')
	const {Element3D, ScrollFling, booleanAttribute} = await import('lume')
	const {autoDefineElements} = await import('lume/dist/LumeConfig.js')

	@element('lume-scroll-item', autoDefineElements)
	class ScrollItem extends Element3D {
		@booleanAttribute skip = false
	}

	@element('lume-scroller', autoDefineElements)
	class Scroller extends Element3D {
		override readonly hasShadow = true
		
		@booleanAttribute scrollX = true
		@booleanAttribute scrollY = true
		
		...

		override template = () => html`
			<lume-element3d size-mode="p p" size="1 1" id="scrollarea">
				...
			</lume-element3d>
		`
	}

	return [Scroller, ScrollItem] as const // <----------------------------- HACK
}

type ScrollerCtor = Awaited<ReturnType<typeof defineElement>>[0] // <----------------------------- HACK
export type Scroller = InstanceType<ScrollerCtor>

declare module 'solid-js' {
	namespace JSX {
		interface IntrinsicElements {
			'lume-scroller': ElementAttributes<Scroller, ScrollerAttributes>
		}
	}
}

declare global {
	interface HTMLElementTagNameMap {
		'lume-scroller': Scroller
	}
}

type ScrollItemCtor = Awaited<ReturnType<typeof defineElement>>[1] // <----------------------------- HACK
export type ScrollItem = InstanceType<ScrollItemCtor>

declare module 'solid-js' {
	namespace JSX {
		interface IntrinsicElements {
			'lume-scroll-item': ElementAttributes<ScrollItem, ScrollItemAttributes>
		}
	}
}

declare global {
	interface HTMLElementTagNameMap {
		'lume-scroll-item': ScrollItem
	}
}
```

</p>
</details> 


The code is now much cleaner with normal imports thanks to https://github.com/ryansolid/dom-expressions/pull/345:

<details><summary>clean code</summary>
<p>

```ts
// ...normal imports...
import html from 'solid-js/html'
import {element, type ElementAttributes, booleanAttribute} from '@lume/element'
import {type Element3DAttributes, Element3D, ScrollFling} from 'lume'
import {autoDefineElements} from 'lume/dist/LumeConfig.js'

@element('lume-scroll-item', autoDefineElements)
class ScrollItem extends Element3D {
	@booleanAttribute skip = false
}

declare module 'solid-js' {
	namespace JSX {
		interface IntrinsicElements {
			'lume-scroll-item': ElementAttributes<ScrollItem, ScrollItemAttributes>
		}
	}
}

declare global {
	interface HTMLElementTagNameMap {
		'lume-scroll-item': ScrollItem
	}
}

@element('lume-scroller', autoDefineElements)
class Scroller extends Element3D {
	override readonly hasShadow = true
	
	@booleanAttribute scrollX = true
	@booleanAttribute scrollY = true
	
	...

	override template = () => html`
		<lume-element3d size-mode="p p" size="1 1" id="scrollarea">
			...
		</lume-element3d>
	`
}

declare module 'solid-js' {
	namespace JSX {
		interface IntrinsicElements {
			'lume-scroller': ElementAttributes<Scroller, ScrollerAttributes>
		}
	}
}

declare global {
	interface HTMLElementTagNameMap {
		'lume-scroller': Scroller
	}
}
```

</p>
</details>

**Comment by ryansolid** [comment]
This should be fixed as of 1.9 now.
